### ISSUE TITLE
macOS 14.0 issues

### ISSUE THREAD

**Issue by zearp** [zearp/Nucintosh#130]
The first beta of macOS Sonoma was released. Download directly from Apple [here](https://swcdn.apple.com/content/downloads/23/44/032-94352-A_DB05J15QWT/4x91v0yzolyiat5cat76ieu0h78aeu3d03/InstallAssistant.pkg).

Apple has dropped support for seemingly all Broadcom devices. So no more native wireless and bluetooth until the Ventura drivers are ported. I don't know what this means for any iServices that all work fine with the current Broadcom cards. Things like extending the screen to an iPad, iCloud integrations, Airdrop and so on.

It seems the end of hackintosh as we know it is ending. At least when it comes to a near vanilla experience. Soon we may have to compromise security and such in order to run macOS at that point for me personally it becomes more interesting to move to Linux instead.

Fedora combined with Gnome and some extensions is very macOS like in my experience. Specially if you only use apps designed with gtk in mind. Everything will look nice and consistent just like macOS.

Anyways, for now let's see how far we can get with Sonoma, likely one of the last macOS versions ever to still support Intel cpu's. My guess is after Sonoma there will be one more release with Intel support and then no more. We'll still receive security updates for a few years so there's enough time to either move to something else or buy serialised badly repairable Apple hardware.

**_Issues:_**
- Broadcom wireless; broken
- Broadcom bluetooth; broken
- Intel wireless; itlwm + Heliport only until AirportItlwm is updated
- Intel bluetooth; broken, needs and updated BluetoothFixup kext

tl;dr unless you want to lose functionality and play with the new beta's stay on Ventura 😅

**Comment by naryfa** [comment]
Any updates? What does it take for Broadcom cards to be ported? Do you think anyone will even try doing it?

**Comment by zearp** [comment]
It will be done and is possible. But it will probably require you to disable some protections. We'll have to wait and see what the wizards at OCLP come up with. Mind you the beta is very fresh, the first beta it was only released a few weeks ago. The relentless release schedule from Apple makes it difficult to support 13.x and 14.x for some devs I'm sure.

Ventura will be the last macOS to natively support any Broadcom wireless and bluetooth. It's a big change by Apple, pretty much all of their older machines use Broadcom. So anyone who wants to run newer macOS will have to either upgrade the wifi (impossible most of the time) or use OCLP or buy a new Mac.

What it takes is usually the kexts from a previous release. You can try it yourself but I wouldn't mess with the beta's to be honest. Last few times I did there were some big changes which made it pretty much a waste of my time to keep testing. So I just wait until it the final is close before testing. I don't want to spend hours trying to get something to work only to have Apple break it again in another beta. It can be fun and educational if you want to do it but this time I don't want do it.

To be honest I've moved some of my NUCs over to Linux already. They perform better, run cooler, use less energy and do pretty much what I need them to do. I bult my own Fedora spin with Gnome which is pretty much macOS for Linux, with some extensions it's very good. Very consistent design they borrow(ed) a lot form macOS 😅

tl;dr once OCLP supports it we can probably use their kexts to get Broadcom going again, not sure about the security impact.

https://github.com/dortania/OpenCore-Legacy-Patcher/issues/1076

**Comment by zearp** [comment]
Oh of course having properly working iCloud on Sonoma will likely not happen due to Broadcom and also Intel not being natively supported. So that part will become hit and miss rather than fully supported. I can now put my iPad down and just move my mouse into it or use it as external display. That kind of stuff might stop working as well. Same for stuff like using your iPhone as webcam and so on.

Well have to wait and see and test what works and if its worth updating. Ventura will continue to work properly and will get security updates for at least a few more years. So there is time to migrate to Linux or buy a Mac. I prefer the former due to Macs being unrepairable. They even removed the lifeboat connector to get your data out if the board dies for some reason. No thanks, Tim.

**Comment by naryfa** [comment]
Waiting out until the final OS release is the way to go.

I've been using Linux since Ubuntu 9.04 so it's my turf. The irreparability (or perhaps now more apt non-upgradability) of Apple devices is pushing me away and I've been slowly transferring my tech life to other platforms. The next several months of developments by hackintosh devs will pretty much make the decision for me, but at this point I'm bound to move. The appeal just isn't there anymore, even though YouTubers would try to make people believe otherwise. 

I'm not surprised Linux runs the NUCs cooler. After all, macOS is now bloated. I mean... just look at the ISO (dmg 😉) size. I wish you could try the case that I use [Cirrus7 Nimbini Extended](https://www.cirrus7.com/wp-content/uploads/2018/03/thumb-400x362.png), but I think I mentioned it before.

I'm of a humble opinion (albeit not necessarily correct) that the hackintosh community somewhat props up Apple's sales, because even though some can't afford a real Mac, they still buy their other devices. If I can't run the macOS anymore, then I don't want the iPhone, because my most important aspect of it was using iMessage at the computer. People would say, buy an older Mac if it's just for that... No, it's just the main reason among many.

I guess we'll see how things develop over the next several months.

**Comment by sebasrock156** [comment]
Bluetooth may being fixed with BluetoothFixup (for both) plus IntelBluetoothFirmware and IntelBTPatcher kexts (for Intel) or  BrcmBluetoothInjector and BrcmPatchRAM kexts (for Broadcom); with **-lilubetaall** boot-arg added and sometimes, in **NVRAM --> 7C436110-AB2A-4BBB-A880-FE41995C9F82**  we should be add [as DATA]: **bluetoothExternalDongleFailed: 00** and **bluetoothInternalControllerInfo: 0000000000000000000000000000** values in our own config.plist.

**Comment by zearp** [comment]
macOS Sonoma beta 2: https://swcdn.apple.com/content/downloads/02/02/032-95852-A_JX472QYN2N/0q2do64a61pmgv1u8j0l3ddyfrods0yvdo/InstallAssistant.pkg

Will be trying this today or tomorrow on my stock test NUC and one of my Broadcom ones. Let's see where things stand and for anything broken apply and test fixes where available.

**Comment by zearp** [comment]
Clean install with latest EFI and beta 2 results in only wifi not working. Bluetooth working out of the box and I could pair a mouse just fine. Will try wireless fixes later. I think we might need to wait until the upstream version of AirportItlwm supports Sonoma. Till then using the plain itlwm kext and Heliport might work. I have not tried that yet but on paper it should work. The itlwm kext just doesn't integrate and shows up as ethernet for macOS itself. Hence the need for another app to connect.

**Comment by zearp** [comment]
I do notice both 13.4.1 and 14.x are rarely clocking the cpu below 1.3ghz, whereas on Linux even when playing Youtube the clocks are like 700-900mhz. It runs a lot cooler there. I will look into changing this without affecting performance. Likely need to tweak or even just remove the Lilu cpu friend plugin. The thing is the NUC comes with a few different cpu's that have different frequencies available. So I guess the best would be to get rid of it and monitor the frequencies used. If we can rely on the stock/vanilal scaling that would be great. When doing light work it should scale to the lowest available clock speed. My idle temps are 10c higher in macOS than Linux on the same machine 😅

**Comment by kababoom** [comment]
What about the energy performance bios setting, does this play a role here? I do usually set them to balanced since I rarely need high performance. 
What tool do you use to monitor the clock frequency?

**Comment by Daasin** [comment]
P.s. There is now a version of Airportitlwm that works with sonoma. Successfully tested on NUC-10.

**Comment by zearp** [comment]
I didn't see it in my release folder after syncing and compiling it today. Do we just use the Ventura target?

**Comment by zearp** [comment]
Oh the devs only posted a binary test version buried in an issue and didn't push any changes to the repo yet. I always compile from source with only the firmware we need. Without that the EFI becomes quite large. The test version is 16mb because it includes firmware for all the supported chipsets.

[AirportItlwm-Sonoma-Preview01.zip](https://github.com/zearp/Nucintosh/files/11952721/AirportItlwm-Sonoma-Preview01.zip)

When they update the sources I will recompile it and push it to the repo.

**Comment by zearp** [comment]
Time to update my test machine from 13.4.1 to 14.0 beta 2 and see how things stand. I'm also trying the wireless driver test build.

![Screenshot 2023-07-05 at 03 14 30](https://github.com/zearp/Nucintosh/assets/63272687/ec0f6176-660d-4f29-b218-73a823d57665)

**Comment by zearp** [comment]
Update completed. Bluetooth works, wireless works. Everything looking good for the stock NUC!

Of course Apple enables some snooping when updating to beta's and it will re-enable some of these after each beta update as well haha.

![Screenshot 2023-07-05 at 03 48 55](https://github.com/zearp/Nucintosh/assets/63272687/1704b4aa-5827-44db-9773-b7b79a95c709)

As far as cpu scaling goes, with cpu friend disabled the lowest frequency on my i5 model is 1.2ghz with cpu friend that's 700-800mhz as it's supposed to be. It just rarely runs at those low clock speeds. So it's best to leave those enabled. I'll have a look at the configuration and other related things like the scheduler.

**Comment by cxrtado** [comment]
Hey zearp! 

Amazing to see you still providing solid support and updates for this aging little box. 

I unfortunately have shorted my internal USB headers. Multiple pci boards have just never shown up under both Linux and Mac for me despite the WiFi working a treat. 

Re: Bluetooth, I’ll likely never get the native handoff features. But will a cheapo “CSR” 4.0 Bluetooth dongle, alongside my BCM94360CS2 work in Sonoma? I know you referenced purchasing a bunch of them before. Just need to be able to use my AirPods regularly.

**Comment by zearp** [comment]
Yeah, the CSR dongles are fine for bluetooth audio and mouse/keyboard. Regarding iServices, it remains to be seen what will work and what not. It might become like Intel where it is hit and miss. I would certainly miss the ability to paste in 2FA codes or forwards calls and such. I will remain on Ventura for those features if they can't be made to work. By the time Ventura stops getting security updates macOS will be Apple silicon only 😅

**Comment by cxrtado** [comment]
Certainly getting to that end of the road with hackintoshing. My little Nuc has been chugging along faithfully. M1 or Linux is my next hop after this. But we will wait to see how far “after this” is 😭

**Comment by zearp** [comment]
I'm most likely moving to Linux or maybe FreeBSD. Really depends on how well it can run on the NUC. There is also Google's Fuchsia OS we can run but I don't want to move from Apple spying a bit to Google spying a lot.

I won't be buying any Apple hardware if only for the fact the storage is integrated in the soc and there's no way to read your data if anything else breaks on the board. It doesn't sit right with me to have it all integrated in the soc, specially storage which will wear out at some point. It's not like the performance gains are so much it would justify losing so much flexibly and repairability over for me.

I do wonder if the storage like usb sticks and micro sd card will become read only once it has worn out. That would be fun to have in your machine, a complete disk full of stuff you can't delete anymore. Would also make it impossible to sell unless you don't care some stranger having access to all your stuff.

When Apple started soldering nand chips to the board they at least added a port you can use to get your data in case something on the board died and you couldn't boot into the OS anymore. But on the new Macs you have nothing. So you need to keep like a real time backup if want a chance to recover from sudden h/w failure. I rather not risk it. I have good backups but a few times already laptops or computers just died on me and being able to take out the disk or nvme and just boot form it or copy your data is invaluable to me personally. No hours of work lost.

I always liked that macOS can boot on anything it supports. So if your NUC breaks you can literally take the disk out put in a usb case and connect it to a real Mac and boot in to your desktop. No need for the EFI. You could also boot it on another NUC or any hack given you have a good EFI for it.

Well, a good ramble but luckily for now we can run the latest macOS without compressing security and don't have to worry yet about migrating.

**Comment by zearp** [comment]
Did some experimenting with cpu friend config and this the key to taming these big clock speeds. It is all about the bias towards either saving power or getting to max performance faster. I'm pretty sure that no matter what you configure if the load is high enough and lasts long enough it will always clock to the max speeds. But it may be beneficial for people to make their config based on their wants. I might change the config included by default to behave more like Linux/Windows where it is bit more conservative. It doesn't really affect performance as far as I can tell. Need to do some more testing. This is the config in the EFI now: 
![Screenshot 2023-07-06 at 06 22 49](https://github.com/zearp/Nucintosh/assets/63272687/a0c31f06-ebed-45d3-9b47-059e4aa96729)

This is the test config attached to this post:
![Screenshot 2023-07-06 at 06 36 05](https://github.com/zearp/Nucintosh/assets/63272687/db9a906b-ba19-45c7-b59f-f76b1dcd6c39)

Both are pretty much idle, just Safari opened and signed into iCloud. The core average dropped like 50% and I don't notice any difference in performance or responsiveness. It also runs a bit cooler. Maybe making a few configs to pick from is also a good idea. Max power savings, max performance and as balanced as I can get it.

More info here: https://github.com/corpnewt/CPUFriendFriend

[CPUFriendDataProvider.kext.zip](https://github.com/zearp/Nucintosh/files/11964178/CPUFriendDataProvider.kext.zip)

**Comment by zearp** [comment]
Just noticed the package power went from 9 to 4 watts, I see it drop as low as 2 haha. Intel took our undervolting away so this may be a good way to keep temps under control without losing performance for passive cooling or a more quiet machine. Might take a bit longer to reach the max clock speeds but I personally don't mind waiting a few ms longer here and there if it reduces power consumption and runs quite a bit cooler.

**Comment by zearp** [comment]
Okay, I cheered too soon! It doesn't scale up all the way but this is promising. Some tweaking of the config and more testing should result in a good config that still turbo's to the max speed while also being more conservative when idle or under low load. I don't want the cpu to idle with a high average clock as seems to be the case in the current EFI 😅

![Screenshot 2023-07-06 at 06 41 50](https://github.com/zearp/Nucintosh/assets/63272687/66bcaa3d-f99a-456c-8ff6-42d92596e66f)

**Comment by zearp** [comment]
The NUC8 has been very good indeed.. reliable little boxes!

A few more years before Linux is my only options for these. By that time who knows what happening. I wouldn't mind some kind of crazy Ampere based system with 512 cores or something lol.

Meanwhile beta 3 was released on the 5th but I didn't get an OTA update of it...

https://swcdn.apple.com/content/downloads/25/58/042-06331-A_J3SK37ZPVW/a6x4qllf9rcgs2phc7mjyo6v1bpqb87anf/InstallAssistant.pkg

**Comment by zearp** [comment]
New wireless driver build. I hope they update the sources soon.

[AirportItlwm-Sonoma-Preview02.zip](https://github.com/zearp/Nucintosh/files/12001102/AirportItlwm-Sonoma-Preview02.zip)

**Comment by zearp** [comment]
@naryfa you can check progress on the OCLP projects issue linked here: https://github.com/zearp/Nucintosh/issues/130#issuecomment-1604983599 -- short answer is no.

**Comment by naryfa** [comment]
OK, thanks for the info.

Read today also that Intel is exiting NUCs. I'm only throwing this in here as a side note, I know it's unrelated, but ... somewhat significant:

https://www.anandtech.com/show/18952/intel-set-to-exit-nuc-pc-business-pushes-partners-to-develop-more-sff-pcs

**Comment by danielHero307** [comment]
Hi, I bought a used Mac because of the Apple ecosystem, I have an iPad and an iPhone, and a Mac laptop can help me, otherwise I had more options than I could buy, especially for installing Linux. I think Apple's security review this project for the new OS Update and they try to disable this project. thanks for all your efforts . ❤️💕

**Comment by stefanmrvic** [comment]
Can I use LG 5k ultrafine 27' monitor on MacOS Ventura on Nucintosh?

**Comment by zearp** [comment]
@stefanmrvic wrong place to ask but it should work. If it connects using Thunderbolt it has to be connected before power on because hot plug doesn't work without patching the firmware and patching the firmware breaks sleep and some usb stuff. If it just uses display port then use the usb-c for it or if using hdmi use the hdmi port which may require some tweaks in the config but it should also just work. I've already enabled all the needed things for 4k but there's additional options we can add if it isn't enough.

I don't have any 5k displays myself but 4k works just fine.

**Comment by stefanmrvic** [comment]
> @stefanmrvic wrong place to ask but it should work. If it connects using Thunderbolt it has to be connected before power on because hot plug doesn't work without patching the firmware and patching the firmware breaks sleep and some usb stuff. If it just uses display port then use the usb-c for it or if using hdmi use the hdmi port which may require some tweaks in the config but it should also just work. I've already enabled all the needed things for 4k but there's additional options we can add if it isn't enough.
> 
> I don't have any 5k displays myself but 4k works just fine.

After seeing the price of LG 5k, i think I'll just go with the 4k ;D

Can you tell me which 4k monitor you're using? I'm thinking of buying LG 24" 24UD58-B or LG 24'' 24MD4KLB-B

You have any recommendations? @zearp

**Comment by zearp** [comment]
I don't know the models I tested the EFI on, it's been a while but 4k was fine. 

Just remember the NUCs HDMI is not really 2.0 it uses some kind of funky business called lspcon. The port is really a display port 1.2a. It is setup right but you may have better results using the usb-c port.

```
HDMI is supported through a MegaChips MCDP2800-BCT DisplayPort 1.2a to HDMI 2.0 Level Shifter/Protocol Converter (LSPCON). The HDMI port supports standard, enhanced, or high definition video, plus multi-channel digital audio on a single cable. The port is compatible with all ATSC and DVB HDTV standards and supports eight full range channels at 24-bit/192 kHz audio of lossless audio formats. The maximum supported resolution is 4096 x 2160 @ 60 Hz, 24bpp. The HDMI port is compliant with the HDMI 2.0a specification.
```

Using usb-c gives you some extras like daisy chaining and some minitors prefer no conversion. The usb-c port is a display port 1.2 without conversion. The HDMI port uses conversion to make a HDMI signal. Some screen do not like that.

```
DisplayPort output supports Multi-Stream Transport (MST) which allows for multiple independent video streams (daisy-chain connection with multiple monitors) over a single DisplayPort. This will require the use of displays that support DisplayPort 1.2 and allow for this feature.
```

Good luck, and as always devices on hackintosh are just a likely or unlikely to work as on real Macs. Which can be pretty finicky. I've even had cables that worked fine on Windows and Linux but didn't support all the resolution or refresh rates in macOS. It can be really picky with certain things. There's nothing we can do about that. Mac users run into the same problems. You don't wanna know how many posts I've seen on Apple's forums about screens and printers just not working properly. Or other strange behaviours. Unless it has this "Mad for Mac" label its a gamble if it works.

**Comment by zearp** [comment]
I don't know the models I tested the EFI on, it's been a while but 4k was fine. 

Just remember the NUCs HDMI is not really 2.0 it uses some kind of funky business called lspcon. The port is really a display port 1.2a. It is setup right but you may have better results using the usb-c port.

5k isn't officially supported but it did work for me. It is likely due to the display port tweaks we apply like ```enable-max-pixel-clock-override```.  There are some other options too that could be applied like setting HBR rates. It won't support full 10bit due to bandwidth limitations so only 8bit colours and maybe at 30hz or 60hz if lucky. Its been voer 3 years since I tested all these things with stuff I had available at work.

```
HDMI is supported through a MegaChips MCDP2800-BCT DisplayPort 1.2a to HDMI 2.0 Level Shifter/Protocol Converter (LSPCON). The HDMI port supports standard, enhanced, or high definition video, plus multi-channel digital audio on a single cable. The port is compatible with all ATSC and DVB HDTV standards and supports eight full range channels at 24-bit/192 kHz audio of lossless audio formats. The maximum supported resolution is 4096 x 2160 @ 60 Hz, 24bpp. The HDMI port is compliant with the HDMI 2.0a specification.
```

Using usb-c gives you some extras like daisy chaining and some minitors prefer no conversion. The usb-c port is a display port 1.2 without conversion. The HDMI port uses conversion to make a HDMI signal. Some screen do not like that.

```
DisplayPort output supports Multi-Stream Transport (MST) which allows for multiple independent video streams (daisy-chain connection with multiple monitors) over a single DisplayPort. This will require the use of displays that support DisplayPort 1.2 and allow for this feature.
```

Good luck, and as always devices on hackintosh are just a likely or unlikely to work as on real Macs. Which can be pretty finicky. I've even had cables that worked fine on Windows and Linux but didn't support all the resolution or refresh rates in macOS. It can be really picky with certain things. There's nothing we can do about that. Mac users run into the same problems. You don't wanna know how many posts I've seen on Apple's forums about screens and printers just not working properly. Or other strange behaviours. Unless it has this "Mad for Mac" label its a gamble if it works.

**Comment by zearp** [comment]
One issue is that I still don't get beta OTA updates offered despite some being available. This is probably easy to fix and might disappear on its own when we near the finals builds. Developer build 4 was released on the 25th and can be downloaded here: https://swcdn.apple.com/content/downloads/51/00/042-16905-A_EL0PGKG0SG/uhav59i5qk1wfy42ild7wscq85tpaqvdws/InstallAssistant.pkg

Other than that every update so far has installed fine and didn't break anything on stock NUCs.

**Comment by MrMatu** [comment]
Hello. I am having trouble installing this BETA on my Mac Mini 2012. It gets stuck on the Apple logo after doing all the process like I am supposed to using OpenCore.

**Comment by zearp** [comment]
@MrMatu Wrong Gitgub repo I. think. But to answer your question; your model is no longer supported by Apple for anything beyond Catalina for a while now. You will need to use OCLP to install Big Sur or newer on your system. Though having some 2012 models myself I wouldn't install anything beyond Catalina on those as the cpu simply lacks a lot of instructions newer macOS take advantage of. So it will fee sluggish. For 2012 models I suggest Fedora + Gnome or Catalina (which no longer received security updates).

**Comment by zearp** [comment]
@MrMatu Wrong Gitgub repo I think?

But to answer your question; your model is no longer supported by Apple for anything beyond Catalina for a while now. You will need to use OCLP to install Big Sur or newer on your system. Though having some 2012 models myself I wouldn't install anything beyond Catalina on those as the cpu simply lacks a lot of instructions newer macOS take advantage of. So it will fee sluggish. For 2012 models I suggest Fedora + Gnome or Catalina (which no longer received security updates).

**Comment by MrMatu** [comment]
I know it is not supported and I was using OCLP already. The mac 14 beta became available a few days ago and ever since I've had no luck installing it. It gets stuck on the LOGO. But Thanks.

**Comment by semakj** [comment]
Everything works fine on my iMac 27" Late 2013 (14,2), Nvidia GEForce GT 755M. However, I cannot boot into Safe Mode with root patches installed. I need Safe Mode to temporarily disable 3D acceleration to be able to perform virgin, initial runs of Acrobat Reader and Speedtest. (Due to som ancient bug in the NVIDIA root patches).

**Comment by zearp** [comment]
For some weird reason my Sonoma test machine's wifi worked fine until today and I had to update the kext to the latest (still no source available) binary itlwm test release. Also their Bluetooth sources don't build anymore. I hope they fix it at some point I much rather compile my own with only needed firmwares than have to include 30-50MB of firmwares we do not need. It may also grow the EFI too large for some of my machines that multi boot into Linux and have a small EFI partition. Sure I can make it bigger but I shouldn't have to if the sources can be built. It also rubs me the wrong way when the provides source es can't be built so we have to download a binary we have no clue whats in it. These kexts run at a very low level giving lots of access...

[AirportItlwm-Sonoma-Preview04.zip](https://github.com/zearp/Nucintosh/files/12270407/AirportItlwm-Sonoma-Preview04.zip)

**Comment by zearp** [comment]
For some weird reason my Sonoma test machine's wifi worked fine until today and I had to update the kext to the latest (still no source available) binary itlwm test release.

Also their bluetooth sources don't build anymore. I hope they fix it at some point I much rather compile my own with only needed firmwares than have to include 30-50MB of firmwares we do not need. It may also grow the EFI too large for some of my machines that multi boot into Linux and have a small EFI partition. Sure I can make it bigger but I shouldn't have to.

It rubs me the wrong way when the provided source can't be build so we have to download a binary we have no clue whats in it. These kexts run at a very low level giving lots of access. Open source should mean we can build the binary using the provides sources but I haven't been able to compile the bluetooth drivers for a while now and I'm not willing to spend a lot of time trying to fix that. There is little point in auditing sources when you have no idea if the binary they provide is build using the same source.

Luckily bluetooth still work as we only use the firmware provider kext, the firmware itself is loaded by another kext. My hope is that we never have to change the OpenIntelWireless bluetooth kext again as it just contains the firmware we need. It is unlikely a newer firmware version is released by Intel for this so for now it's all good.

[AirportItlwm-Sonoma-Preview04.zip](https://github.com/zearp/Nucintosh/files/12270424/AirportItlwm-Sonoma-Preview04.zip)

**Comment by KirsebaerQAQ** [comment]
Broadcom fixed now?

**Comment by zearp** [comment]
@KirsebaerQAQ No. Apple removed support for all Broadcom from Sonoma. We have two wait for OCLP to sort it and then see if we can back port their solution. I posted al ink to the relevant thread earlier in here if you want to keep to to date with Broadcom fixes follow that.

Stock NUCs work 100% in Sonoma including Intel bluetooth and wifi.

**Comment by zearp** [comment]
For those interested this is the EFI I am currently testing with. 

[EFI-Sonoma.zip](https://github.com/zearp/Nucintosh/files/12294065/EFI-Sonoma.zip)

**Comment by naryfa** [comment]
This is hilarious. Stock WiFi/BT from Intel works, but the original one from Apple - nope. 

I remember the same situation with lossless Apple Music. It works on Windows and Android, but not older macOS.

**Comment by zearp** [comment]
New beta dropped last week, again I didn't get an update offered for it. I'll do a clean install on my test NUC soon and see if this can be fixed or needs to be postponed until the GM release.

Developer beta 5: https://swcdn.apple.com/content/downloads/26/36/042-27162-A_4GKRLRWELJ/qyzyo86g692wlsewkyclfk686op47kuq5c/InstallAssistant.pkg

**Comment by zearp** [comment]
Updated EFI for beta 5 and above. It is required to either use this EFI to update or disable AirportItlwm in the config to prevent a bootloop during install.

[EFI.zip](https://github.com/zearp/Nucintosh/files/12371036/EFI.zip)

**Comment by treforparr** [comment]
Can we please try and fix thunderbolt two driver any developers willing to pay to fix this

**Comment by zearp** [comment]
@treforparr the only thing "broken" is hot plug which will not only requires you to flash a new bios into the controller but also need to write new acpi patches. I experimented with it but it's not worth the time and effort to be honest. Thunderbolt devices are very costly for testing purposes and all this just to get hot plug to work is simply not worth it. Indeed your best bet is to hire someone to do this. Thunderbolt works great, all devices just no hot plugging.

**Comment by zearp** [comment]
Developer beta 6 (public beta 4) was releases yesterday. Again no OTA update being offered. No update is even found when checking manually:

```
$ softwareupdate -l
Software Update Tool
Finding available software
No new software available.
```

I will do a clean install of this one and then see if we get the OTA for beta 7 offered or not. Here is is the direct download link to the installer: https://swcdn.apple.com/content/downloads/44/25/042-37835-A_BEAOB8L7HU/1qzvpcz2hplmg557cci42sr49egqus76v1/InstallAssistant.pkg

**Comment by zearp** [comment]
Clean install went ok, restoring my stuff also went ok using the migration tool. I assume the OTA updates will work eventually. I recall similar issues on previous beta's.

**Comment by zearp** [comment]
Developer beta 7 / public 5 has been released a few days ago: https://swcdn.apple.com/content/downloads/14/54/042-41491-A_WCZEM7L2US/b5eeeylcnmf82ycto51pvy0klcyzd02on8/InstallAssistant.pkg

Will download and update my cleanly installed Sonoma from 2 weeks ago. Will update here if any changes to the EFI are needed.

**Comment by zearp** [comment]
Booted up the Sonoma disk to test if I got offered an OTA this time and noticed a little warning saying I should sign into Apple ID to receive updates. I hope this isn't going to be in the final version. 

Currently we get updates regardless if you're signed in or not. I'm not signed in on any of my test machines and I won't sign it as well and I also don't want to make a throw away Apple ID to test this.

I'll assume that by the time Sonoma is released auto updates will " justwork" and I just hope they will drop the sign in requirement to receive updates.

**Comment by zearp** [comment]
Lovely, even after checking the SHA sums and verified my download was proper it still proceeded to download the full 12GB installer again when I ran the installer.

```
Sep  4 18:07:16 nuc osinstallersetupd[961]: mountDiskImageWithPath: /Applications/Install macOS Sonoma beta.app/Contents/SharedSupport/SharedSupport.dmg
Sep  4 18:07:16 nuc osinstallersetupd[961]: Couldn't mount dmg! (error code 13)
```

This isn't the first time this happened. What a waste of bandwidth lol.

**Comment by zearp** [comment]
Well, no changes to the EFI posted here earlier are needed. This build introduces quite some animated slo-mo wallpapers in addition to the previously introduced widgets. Not of a fan of the direction Apple is taking macOS. But the writing has been on the walls since Big Sur.

![Screenshot 2023-09-04 at 20 55 51](https://github.com/zearp/Nucintosh/assets/63272687/20893439-f421-4f76-82db-93d8e5308b03)

**Comment by Lorys89** [comment]
@zearp hello, so if you want I can confirm that the cfg lock can be unlocked, and that the card removal icon can be removed from the title bar and that hotplug tb can be fixed without fw flash.

**Comment by zearp** [comment]
@Lorys89 Feel free to share.

**Comment by Lorys89** [comment]
> @Lorys89 Feel free to share.

Send me your acpi origin estra
Send me your acpi origin Or sysreport

**Comment by zearp** [comment]
RC1 coming soon and final scheduled for the 26th this month. When the RC comes out it's time to check on the state of Broadcom. I'm also pretty sure Sonoma will be the last macOS for the Mac mini 2018. Not sure if Apple will drop native 8th gen support soon but they will at some point. Haswell (4th gen) was dropped after Monterey.

**Comment by naryfa** [comment]
I'd love to see Broadcom ported back, otherwise I have to take my unit apart and remove it. But then I'd get the m.2 port back, so maybe it's a better solution. Does Intel WiFi/BT work good with the current kexts?

**Comment by zearp** [comment]
Unless something magical happens Intel would be the way to go in Sonoma. Currently the only way to make Broadcom work --and that is just the networking, not the iCloud/etc stuff-- is to disable many security features, including breaking the filesystem seals. Which might break receiving updates and having to manually install those combo updates of yesteryear when and if they come out. Bit all that needs to be tested. It is not looking good for Broadcom. 

Intel can be used without disabling any security features. I don't think that even with the patches and security disabled the iCliud integration will work as before. Some stuff might work but it is no longer natively supported. Which was the only reason to even consider Broadcom cuz Intel was working for a long time now with itlwm. You just didn't get proper iCloud integration. Broadcom will be on par with that with extra downsides.

My initial goal was to run vanilla macOS as its intended. That includes all the security features, it is part of what makes is so stable. Even by accident a user or piece of software can mess up your system but if we start breaking and disabling those things it may result in problems or worse, your machine being hacked and you wouldn't even know about it as the underlying security mechanism are disabled or even removed.

I'm not sure if and how we integrate those patches. Most likely we need to create an installer as before, add the EFI and then patch the installer with OCLP. Once installed the system itself also needs to be patched. Whenever a macOS update comes out we have to make sure it won't break any of the patches. Unlike currently where you can just pretend you have a real Mac pretty much. Once macOS is released we have to see whats going to happen and what the Dortania guys will do. Currently the OCLP does not support using it on hackintosh. But many many people have Broadcom in their hacks. So we'll see. Maybe some other solution is found.

tl;dr
Broadcom is most likely EOL with Sonoma, at least for me.

**Comment by naryfa** [comment]
Understood. Does iCloud integration work OK with Intel now? You mentioned that it didn't, as in past tense, has anything changed?

**Comment by zearp** [comment]
I haven't tested and can't test all the continuity stuff but last time I tried I couldn't use my iPad as external display, something that goes flawless with Broadcom. Same for moving the mouse/keyboard to the iPad. Not sure how that feature is called. I think AirPlay worked but AirDrop didn't. Same as automatically pasting in 2FA codes did not work on Intel. It's a real mixed bag.

I'll start testing soon but won't update the EFI until there's more clarity regarding Broadcom. For now the current EFI works well for stock NUCs  and Sonoma. Vanilla NUC users will have no issues at all. It is a bit annoying for me personally because my main NUC in its passive case has no antenna connectors on the Intel chip anymore. They snapped off, both of them when very carefully removing them to put the board in the passive case. So if I remove Broadcom I won't have any wireless. Very difficult to solder on a new connector. I'm not bad at soldering but this is above my pay grade.

I did see some Chinese users desolder the Intel and also desolder the card reader chips and re-using those pci busses for other stuff. Pretty cool hacks but I lack the skills for that lol. I hope someone finds a way to get Broadcom to work without breaking the iCloud stuff or the need to disable a lot of security features.

**Comment by amiralimahmoudi** [comment]
I've found a big issue for using OCLP to bring back my wifi
after I've done the settings mentioned in my config.plist before even run OCLP creative cloud app and WhatsApp does not open anymore, and when I discard my config.plist and bring it to back everything work just fine
need to mentioned that I've successfully launched wifi, but cause of that problem had to change my config.plist to defualt

**Comment by zearp** [comment]
@amiralimahmoudi I have read several times but I have no idea what you meant. Sorry.

**Comment by amiralimahmoudi** [comment]
> @amiralimahmoudi I have read several times but I have no idea what you meant. Sorry.

Hi there, my Wi-Fi card ( Archer T9) is not supported in Sonoma anymore.
there is an instruction using OCLP which can bring back the Wi-Fi.
there is some instruction that should be done with the config.plist like csr config and.... ok? after doing those instructions to my config.plist (before launching OCLP 0.6.9) after restarting and resetting NVRAM when my hackintosh comes up the Creative Cloud App keeps crashing and not working anymore so I cannot use Creative Cloud app or Adobe extensions. but when I bring back my previous config.plist creative cloud works just fine. the problem is not OCLP application cause this issue happens before launching OCLP, and I'm pretty sure doing all instructions correctly because I enabled my wifi

**Comment by wongkarc** [comment]
Regarding the macbook pro 2017 amd radeon pro 560 graphics card that cannot be driven in the sonoma system, is there a way to make it drive?

**Comment by zearp** [comment]
Please remember this issue is not for generic support or help. It is for Sonoma on our NUC8 computers using the repo's EFI. There are much better places to get help with those issues.

**Comment by naryfa** [comment]
> I haven't tested and can't test all the continuity stuff but last time I tried I couldn't use my iPad as external display, something that goes flawless with Broadcom. Same for moving the mouse/keyboard to the iPad. Not sure how that feature is called. I think AirPlay worked but AirDrop didn't. Same as automatically pasting in 2FA codes did not work on Intel. It's a real mixed bag.
> 
> I'll start testing soon but won't update the EFI until there's more clarity regarding Broadcom. For now the current EFI works well for stock NUCs and Sonoma. Vanilla NUC users will have no issues at all. It is a bit annoying for me personally because my main NUC in its passive case has no antenna connectors on the Intel chip anymore. They snapped off, both of them when very carefully removing them to put the board in the passive case. So if I remove Broadcom I won't have any wireless. Very difficult to solder on a new connector. I'm not bad at soldering but this is above my pay grade.
> 
> I did see some Chinese users desolder the Intel and also desolder the card reader chips and re-using those pci busses for other stuff. Pretty cool hacks but I lack the skills for that lol. I hope someone finds a way to get Broadcom to work without breaking the iCloud stuff or the need to disable a lot of security features.

Mine's also passive, but the connectors are fine. I just don't want to lose the functionality by using Intel.

**Comment by amiralimahmoudi** [comment]
> @amiralimahmoudi
> 
> to fix the apps that do not work ,use the amfi kext instead of the boot argument. https://github.com/dortania/OpenCore-Legacy-Patcher/blob/sonoma-development/payloads/Kexts/Acidanthera/AMFIPass-v1.3.1-RELEASE.zip

I've added amfi kext into my EFI folder and your solution fixed creative cloud problem, but after launching OCLP and rooting, the system is not booting and it stuck on apple logo and in th verbose it says AMFI ( sstem/Library..... I guess) but when I revert rooting,  system works fine but no WIFI because cant use OCLP rooting

**Comment by zearp** [comment]
Tomorrow the GM will be released, I'll be updating the EFI and doing a clean install on a stock NUC. If that all works as expected I'll focus on Broadcom and see how far we can get in making it run without crippling security. This issue has served it's purpose.

**Comment by zearp** [comment]
Final version has been released: https://swcdn.apple.com/content/downloads/26/09/042-58988-A_114Q05ZS90/yudaal746aeavnzu5qdhk26uhlphm3r79u/InstallAssistant.pkg
